src/system/threader.ts

```diff
@@ -74,6 +74,14 @@ export class Threader {
         const authorText = `${message.author.username}#${message.author.discriminator}`;
         const threadName = `${authorText} Ticket`;
         const author = `<@!${message.author.id}>`;
+        const userRoles = message.member?.roles.cache.map((role) => role.id);
+        
+        if(!userRoles?.some((role) => config.rolesAllowedToUse.includes(role))) {
+            await message.channel.send(config.errorMessage);
+            await message.delete();
```

Comment:
Deelte one message.delete() and put the other one above the check to reduce duplicated code